spec/reek/smells/attribute_spec.rb

```diff
@@ -17,6 +17,47 @@
       ctx = Reek::Core::CodeContext.new(nil, Reek::Source::SourceCode.from(src).syntax_tree)
       expect(@detector.examine_context(ctx)).to be_empty
     end
+
+    shared_examples_for 'no attribute found' do
+      before :each do
+        ctx = Reek::Core::CodeContext.new(nil, Reek::Source::SourceCode.from(@src).syntax_tree)
+        @smells = @detector.examine_context(ctx)
+      end
+
+      it 'records no attribute' do
+        expect(@smells.length).to eq(0)
+      end
+    end
+
+    context 'declared in a class' do
```

Comment:
Can you make these context a bit more specific? You now have to read the actual value assigned to `@src` to understand what's being tested.